File: MacApp/MacWorkspaceWindow.vb
Imports AppKit
Imports Foundation

Namespace Application

  Public Class MacWorkspaceWindow
    Inherits NSWindow

    Public Sub New(contentRect As NSRect, styleMask windowStyle As NSWindowStyleMask, backing backingStore As NSBackingStoreType, defer shouldDefer As Boolean, screen targetScreen As NSScreen)
      MyBase.New(ContentRect: contentRect,
        styleMask: windowStyle,
        backing: backingStore,
        defer: shouldDefer,
        screen: targetScreen)
    End Sub

    Public Overrides Function canBecomeKeyWindow() As Boolean
      Return True
    End Function

    Public Overrides Function canBecomeMainWindow() As Boolean
      Return True
    End Function

  End Class

End Namespace

File: MacApp/MacWorkspaceWindow.pas
namespace MacApp;

uses
  AppKit,
  Foundation;

type
  MacWorkspaceWindow = public class(NSWindow)
  public
    constructor withContentRect(contentRect: NSRect) styleMask(windowStyle: NSWindowStyleMask) backing(backingStore: NSBackingStoreType) defer(shouldDefer: Boolean) screen(targetScreen: NSScreen);
    begin
      inherited constructor withContentRect(contentRect) styleMask(windowStyle) backing(backingStore) defer(shouldDefer) screen(targetScreen);
    end;
    method canBecomeKeyWindow: Boolean; override;
    begin
      result := true;
    end;
    method canBecomeMainWindow: Boolean; override;
    begin
      result := true;
    end;
  end;

end.
